llmfoundry/data/finetuning/tasks.py

```diff
@@ -109,37 +109,42 @@ def _is_empty_or_nonexistent(dirpath: str) -> bool:
 
 def _get_role_key(message: Dict[str, str]) -> str:
     role_keys = _ALLOWED_ROLE_KEYS.intersection(message.keys())
-    assert len(role_keys) == 1
+    assert len(
+        role_keys) == 1, f'Expected 1 role key, but found {len(role_keys)}'
     role_key = list(role_keys)[0]
     return role_key
 
 
 def _get_content_key(message: Dict[str, str]) -> str:
     content_keys = _ALLOWED_CONTENT_KEYS.intersection(message.keys())
-    assert len(content_keys) == 1
+    assert len(content_keys
+              ) == 1, f'Expected 1 content key, but found {len(content_keys)}'
     content_key = list(content_keys)[0]
     return content_key
 
 
 def _get_message_key(example: ChatFormattedDict):
```

Comment:
I think this doesn't cover the error I sent you, which results from the data not matching the expected types. Since this is data validation, lets not rely on the types being correct, and do some runtime checking to raise nice errors when data is malformatted